tests/lib/index.js

```diff
@@ -1063,5 +1063,481 @@ describe('config', function () {
                 );
             });
         });
+
+
+        /* jshint ignore:start */
+        describe("readPromises()", function() {
+          it("fails on unknown bundle", function(next) {
+            var config = new Config();
+            config.readPromise("foo", "bar", {}).catch(err => {
+              expect(err.message).to.equal('Unknown bundle "foo"');
+              next();
+            });
+          });
+
+          it("fails on unknown config", function(next) {
+            var config = new Config();
+            config.addConfig(
```

Comment:
can these tests use `config.addConfigPromise` to avoid the callback pyramid?

Reply:
Updated.